test/hooks/useSelector.spec.js

```diff
@@ -246,6 +246,39 @@ describe('React', () => {
 
           expect(renderedItems.length).toBe(1)
         })
+
+        it('calls selector exactly once on mount and on update', () => {
```

Comment:
:hand: can you also add a test that does force the selector to run a second time?  This should be doable by rendering a child component inside of a parent, and dispatching inside of a `useLayoutEffect` in the child. They mount in the same pass, and the child's `useLayoutEffect` should run before the parent's does.